### ISSUE TITLE
Add acceptance tests for user-sync OCS API

### ISSUE THREAD

**Issue by phil-davis** [owncloud/core#36236]
The user-sync OCS API was added in PR #36216 

Add API acceptance tests for it.

**Comment by phil-davis** [comment]
I added this to Backlog for planning in the current sprint, so that it gets noticed and someone can decide to schedule it appropriately for next week or...

**Comment by phil-davis** [comment]
To test this in core acceptance tests, we would need to have the user_ldap app installed-enabled-configured.
So we can more easily test the end-point in the user_ldap app. That will at least give us nightly feedback ffrom the user_ldap test run.

**Comment by skshetry** [comment]
@phil-davis, so, transfer this to `user_ldap`?

**Comment by dpakach** [comment]
Some tests for user-sync API are added in the PR https://github.com/owncloud/core/pull/36428. That only covers the cases when we expect the API to fail. (eg. using the api by non admin user, using the api with wrong password etc.)

In order to test the cases when the user-sync is expected to pass we would need a user backend such as LDAP. That is being blocked by https://github.com/owncloud/core/issues/36312

**Comment by micbar** [comment]
@dpakach What is the status of this ticket? Still needed in the 10.4.0 Context?

**Comment by phil-davis** [comment]
We never managed to work out how to successfully send a `curl` request to the new user-sync API end point, with our "usual" LDAP server docker image etc in place like in the user_ldap app acceptance tests. So there is no automated test for the "happy path" of this new end-point.

If someone can demonstrate "manually" how to setup LDAP backend etc so that the end-point works, then we can easily automate it.

For 10.4.0 testing, someone needs to be able to test this manually anyway.

**Comment by dpakach** [comment]
> @dpakach What is the status of this ticket? Still needed in the 10.4.0 Context?

yes just like @phil-davis said, we never managed to get it to work. It would be nice if someone demonstrated how to do it.

**Comment by micbar** [comment]
I vote for closing.

**Comment by DeepDiver1975** [comment]
**Classification:** invalid-fixed (already implemented)

Acceptance tests for the user-sync OCS API now exist on `master` and are active:

- `tests/acceptance/features/apiMain/userSync.feature` — a `@api` feature with scenarios covering: syncing a user when there is no external backend, syncing a non-existent user, syncing as a non-existent user, syncing as a non-admin (`@smokeTest`), and syncing with an incorrect admin password.
- Backed by step definitions in `tests/acceptance/features/bootstrap/FeatureContext.php` (e.g. `@When the administrator tries to sync user :user using the OCS API` and related steps).

These were introduced by #36428 (merged 2019-12-20) and the feature file remains present and maintained on `master`.

As discussed in this thread, the LDAP-backed "happy path" success case was never automated because nobody could reliably reproduce the working end-point manually with the test LDAP setup — and @micbar voted to close. The originally-requested test coverage (the OCS-level cases the end-point can express without a live external backend) is in place, so closing this as resolved.

If the LDAP-backed happy-path automation is still wanted, please open a fresh, scoped issue describing the manual setup needed to make the end-point succeed.

🤖 Automated issue analysis by Claude Code review agent.
